examples/jre/src/main/java/com/fiskaly/sdk/demo/jre/Main.java

```diff
@@ -2,13 +2,297 @@
 
 import com.fiskaly.sdk.*;
 
+import java.io.IOException;
+import java.util.Collections;
+import java.util.UUID;
+
+import com.fiskaly.sdk.factories.GsonFactory;
+import com.google.gson.Gson;
+
 public class Main {
+  static String tssUUID = "";
+  static String clientUUID = "";
+  static String adminPUK = "";
+  static FiskalyHttpClient client = null;
+  static String adminPIN = "0123456789";
+  static String transactionUUID = "";
+  static int transactionRevision = 0;
+  private static final Gson GSON = GsonFactory.createGson();
+
   public static void main(String[] args) throws Exception {
     final String apiKey = System.getenv("FISKALY_API_KEY");
     final String apiSecret = System.getenv("FISKALY_API_SECRET");
-    final FiskalyHttpClient client =
-        new FiskalyHttpClient(apiKey, apiSecret, "https://kassensichv.io/api/v1");
+    client =
+            new FiskalyHttpClient(apiKey, apiSecret, "https://kassensichv.io/api/v1");
+    listTSS();
+    createTSSV1();
+    createClient();
+    createTransactionV1();
+    finishTransactionV1();
+
+      //now do something similar on v2
+    final String apiKeyv2 = System.getenv("FISKALY_API_KEY_V2");
+    final String apiSecretv2 = System.getenv("FISKALY_API_SECRET_V2");
+    //set these to use an existing TSS instead of creating a new one
+    final String existingTSS = System.getenv("FISKALY_TSS_UUID_V2");
+    //set either a PIN (if the TSS already has one set) or a PUK (if the TSS doesn't have a PIN set)
+    final String existingTSSAdminPIN = System.getenv("FISKALY_TSS_ADMIN_PIN");
+    final String existingTSSPUK = System.getenv("FISKALY_TSS_PUK");
+    client =
+        new FiskalyHttpClient(apiKeyv2, apiSecretv2, "https://kassensichv.fiskaly.com/api/v2", "https://kassensichv-middleware.fiskaly.com");
+    listTSS();
+    final Boolean useExistingTSS = existingTSS != null && !existingTSS.isEmpty();
+    if (!useExistingTSS) {
+      createTSS();
+    } else {
+      tssUUID = existingTSS;
+      adminPUK = existingTSSPUK;
+      System.out.println("Using existing TSS '"+existingTSS+"' with PIN '"+existingTSSAdminPIN+"' and PUK '"+existingTSSPUK+"'");
+    }
+    if (!useExistingTSS || existingTSSAdminPIN == null || existingTSSAdminPIN.isEmpty()) {
+      personalizeTSS();
+      changeAdminPIN();
+    } else if (useExistingTSS) {
+      adminPIN = existingTSSAdminPIN;
+    }
+    authenticateAdmin();
+    initializeTSS();
+    createClient();
+    updateClient();
+    createTransaction();
+    updateTransaction();
+    finishTransaction();
+    //don't disable a TSS we didn't create
+    if (existingTSS == null) {
```

Comment:
`existingTSS` is a TSS UUID you set with the `FISKALY_TSS_UUID_V2` environment variable. If you set that (e.g. because there is some problem creating new TSS so you need to use an existing one; I added this code while it was temporarily not possible to create new TSS), then this code uses that existing TSS instead of creating its own, so it doesn't need to disable it. If you don't set `FISKALY_TSS_UUID_V2` then the code creates a TSS at the beginning and disables it at the end. The idea is that after the whole thing runs, there should be the same number of active TSS as before it started.

`disableTSS` disables the TSS `tssUUID`, which will either be the UUID from `existingTSS`, or the UUID of the new TSS that the code created for you because you didn't give it an existing one to use. So it will succeed whether `existingTSS` is null or not.